http/src/handler.rs

```diff
@@ -18,12 +18,38 @@ pub struct PanicHandler {
 	pub handler: Arc<Mutex<Option<Box<Fn() -> () + Send + 'static>>>>
 }
 
+/// RPC Requests handler with additional metadata support.
+pub trait RpcHandler: Send + Sync {
+	/// Type of metadata
+	type Metadata: Send;
+
+	/// Read the metadata from the request
+	fn read_metadata(&self, &server::Request<HttpStream>) -> Option<Self::Metadata> {
+		None
+	}
+
+	/// Handle request with additional metadata.
+	fn handle_request<H>(&self, request_str: &str, response_handler: H, meta: Option<Self::Metadata>) where
+		H: ResponseHandler<Option<String>, Option<String>> + 'static;
+}
+
+impl RpcHandler for IoHandler {
+	type Metadata = ();
+
+	fn handle_request<H>(&self, request_str: &str, response_handler: H, _meta: Option<Self::Metadata>) where
+		H: ResponseHandler<Option<String>, Option<String>> + 'static
+	{
+		GenericIoHandler::handle_request(self, request_str, response_handler)
+	}
+}
```

Comment:
The formatting is consistent with the whole `http` crate:
https://github.com/ethcore/jsonrpc/blob/master/http/.editorconfig

Formatted using tabs, but probably Github is taking into account `tab_width` defined in `.editorconfig`, I can make a separate PR to unify formatting across all crates.